src/bin/pg_autoctl/supervisor.c

```diff
@@ -219,6 +219,9 @@ supervisor_loop(Supervisor *supervisor)
 						log_info("Internal subprocesses are done, stopping");
 						return true;
 					}
+
+					log_fatal("Unexpected ECHILD error from waitpid()");
+					return false;
```

Comment:
It could be that someone did `kill -TERM` or `kill -QUIT` all our child processes without signaling the supervisor process. Now, we could restart every sub-process then, or yeah, bail out to the caller (systemd in many cases) for them to do something about it (restart the main `pg_autoctl` service in many cases).

Reply:
So I don't think we should return false in that case, because we don't know if we still have child process running or not. When we quit and child process are running, they get re-assigned to init (pid 1) on Unix, and having a stale running Postgres in the background is not going to help anyone.

The error message is now a `log_debug()` and we continue looping.